common/qeidentity.c

```diff
@@ -0,0 +1,190 @@
+        // Assert that the qe report's MRSIGNER matches Intel's quoting
+        // enclave's mrsigner.
+        if (!oe_constant_time_mem_equal(
+                qe_report_body->mrsigner, g_qe_mrsigner, sizeof(g_qe_mrsigner)))
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        if (qe_report_body->isvprodid != g_qe_isvprodid)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        if (qe_report_body->isvsvn < g_qeisvsvn)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        // Ensure that the QE is not a debug supporting enclave.
+        if (qe_report_body->attributes.flags & SGX_FLAGS_DEBUG)
+            OE_RAISE(OE_VERIFY_FAILED);
+
+        result = OE_OK;
+        goto done;
+    }
+    OE_CHECK(result);
+
+    // Use QE Identity info to validate QE
+    // Check against fetched qe identityinfo
+    OE_TRACE_INFO("qe_identity.issuer_chain:[%s]\n", qe_id_args.issuer_chain);
+    pem_pck_certificate = qe_id_args.issuer_chain;
+    pem_pck_certificate_size = qe_id_args.issuer_chain_size;
+
+    // validate the cert chain.
+    OE_CHECK(
+        oe_cert_chain_read_pem(
+            &pck_cert_chain, pem_pck_certificate, pem_pck_certificate_size));
+
+    // parse identity info json blob
+    OE_TRACE_INFO("*qe_identity.qe_id_info:[%s]\n", qe_id_args.qe_id_info);
+    OE_CHECK(
+        oe_parse_qe_identity_info_json(
+            qe_id_args.qe_id_info, qe_id_args.qe_id_info_size, &parsed_info));
+
+    // verify qe identity signature
+    OE_TRACE_INFO("Calling oe_verify_tcb_signature\n");
+    OE_CHECK(
+        oe_verify_tcb_signature(
+            parsed_info.info_start,
+            parsed_info.info_size,
+            (sgx_ecdsa256_signature_t*)parsed_info.signature,
+            &pck_cert_chain));
+    OE_TRACE_INFO("oe_verify_tcb_signature succeeded\n");
+
+    // Assert that the qe report's MRSIGNER matches Intel's quoting enclave's
+    // mrsigner.
+    if (!oe_constant_time_mem_equal(
+            qe_report_body->mrsigner,
+            parsed_info.mrsigner,
+            sizeof(parsed_info.mrsigner)))
+    {
+        dump_info(
+            "parsed_info.mrsigner:",
+            parsed_info.mrsigner,
+            sizeof(parsed_info.mrsigner));
+        dump_info(
+            "qe_report_body->mrsigner:",
+            qe_report_body->mrsigner,
+            sizeof(qe_report_body->mrsigner));
+        OE_RAISE(OE_VERIFY_FAILED);
+    }
+
+    if (qe_report_body->isvprodid != parsed_info.isvprodid)
+    {
+        OE_TRACE_INFO(
+            "qe_report_body->isvprodid = %x\n", qe_report_body->isvprodid);
+        OE_TRACE_INFO("parsed_info.isvprodid = %x\n", parsed_info.isvprodid);
+        OE_RAISE(OE_VERIFY_FAILED);
+    }
+
+    if (qe_report_body->isvsvn < parsed_info.isvsvn)
+    {
+        OE_TRACE_INFO("qe_report_body->isvsvn = %x\n", qe_report_body->isvsvn);
+        OE_TRACE_INFO("parsed_info.isvsvn = %x\n", parsed_info.isvsvn);
+        OE_RAISE(OE_VERIFY_FAILED);
+    }
+
+    if (qe_report_body->miscselect != parsed_info.miscselect)
```

Comment:
Per Pushkar, this is not needed.

Reply:
I will updated this part of logics according to the Intel instructions I received earlier today.

Reply:
updated